### PR TITLE
QA-1550: Migrate project from master to main

### PR DESCRIPTION
**Pull request by ichengchang**
Jira ticket [QA-1550](https://broadworkbench.atlassian.net/browse/QA-1550).

Here's the artifacts from `master` to be merged into `main`.

### CODE REVIEW

File: .github/workflows/build-backend-function.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        run: chmod +x gradlew
+
+      # Set up ADC for credentials
+      # Ideally this step should be part of action.yml but
+      # GHA does not yet support action within action so we're stuck.
+      # Please refer to https://github.com/actions/runner/pull/612 for updates.
+      - uses: google-github-actions/setup-gcloud@master
+        with:
+          service_account_email: ${{ secrets.DEPLOY_CF_SA_EMAIL }}
+          service_account_key: ${{ secrets.DEPLOY_CF_SA_KEY_JSON }}
```
Comment (davidangb) [inline .github/workflows/build-backend-function.yml:42]: have all the secrets stored in this repo been vetted/approved by devops and/or appsec?

**Review by davidangb — COMMENTED** [review]
one question inline, else looking good for compliance/security-level review. I understand that business logic and Java-level changes will be significantly updated in upcoming PRs, so I didn't dig much on those.

File: .github/workflows/build-backend-function.yml (hunk 1 above)
Comment (ichengchang) [reply .github/workflows/build-backend-function.yml:42]: There are no formal approval yet by `devops` or `appsec` on the compliance/security-level of these keys. These secrets are for GCP `terra-kernel-k8s` only.

Comment (ichengchang) [reply .github/workflows/build-backend-function.yml:42]: The creation on the BQ tables in Test Runner is different from creation of snapshots in `Delta Layer`.  Let's have appsec weighs in and see what's recommended to meet their standards.

**Review by sagehen03 — COMMENTED** [review]
I'd say delete both GCSEvent and GcsBQ since they will be superseded in short order.  Alternatively, I can open a PR later to remove them.

**Comment by davidangb** [comment]
ok, I will hold review until you get answers from appsec, thank you!

**Review by davidangb — APPROVED** [review]
:+1:, assuming a follow-on action to audit secrets

**Review by sagehen03 — APPROVED** [review]

**Pull request merged**